extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/K8sDruidNodeDiscoveryProvider.java

```diff
@@ -270,9 +270,17 @@ private void keepWatching(String namespace, String labelSelector, String resourc
               if (item != null && item.type != null && item.object != null) {
                 switch (item.type) {
                   case WatchResult.ADDED:
+                    if (item.object.getNode().getServices() == null || item.object.getNode().getServices().isEmpty()) {
+                      LOGGER.warn("no services found on [%s] druid [%s] node [%s], skipping", item.type, item.object.getNode().getNodeRole(), item.object.getNode().getDruidNode().getHostAndPort());
+                      break;
+                    }
                     baseNodeRoleWatcher.childAdded(item.object.getNode());
                     break;
                   case WatchResult.DELETED:
+                    if (item.object.getNode().getServices() == null || item.object.getNode().getServices().isEmpty()) {
+                      LOGGER.warn("no services found on [%s] druid [%s] node [%s], skipping", item.type, item.object.getNode().getNodeRole(), item.object.getNode().getDruidNode().getHostAndPort());
+                      break;
+                    }
                     baseNodeRoleWatcher.childRemoved(item.object.getNode());
                     break;
                   default:
```

Comment:
I assume this means there are other `item.type` values that we don't care about.  Maybe a comment explaining that?

Reply:
I added the comment.

I know of at least one other type, MODIFIED.  These are essentially k8s events.